clang/lib/CodeGen/CGHLSLRuntime.cpp

```diff
@@ -701,3 +808,95 @@ void CGHLSLRuntime::emitInitListOpaqueValues(CodeGenFunction &CGF,
     }
   }
 }
+
+std::optional<LValue> CGHLSLRuntime::emitResourceArraySubscriptExpr(
```

Comment:
Could this function be better named? I don't think it emits a resource array subscript expr?

Reply:
I believe the name is quite descriptive - it is emitting code for subscript expression on resource arrays, or in other words,  emitting code for `ArraySubscriptExpr` AST node on a resources array. I think it matches the existing patterns of `CGF.Emit*`, such as `CGF.EmitScalarExp` or `CGF.EmitCallExpr`. Do you have any other suggestions?